src/psyclone/dynamo0p3.py

```diff
@@ -8759,6 +8760,50 @@ def proxy_declaration_name(self):
             return self.proxy_name+"("+str(self._vector_size)+")"
         return self.proxy_name
 
+    def psyir_expression(self):
+        '''
+        Looks up or creates a suitable Symbol for this kernel argument. If
+        the argument is a scalar that has been provided as a literal (in the
+        Algorithm layer) then the PSyIR of the expression is returned.
+
+        :raises NotImplementedError: if this argument is not a literal, scalar
+                                     or field.
+        '''
+        symbol_table = self._call.scope.symbol_table
+
+        if self.is_literal:
+            reader = FortranReader()
+            return reader.psyir_from_literal_expression(self.name)
+
+        elif self.is_scalar:
+            try:
+                scalar_sym = symbol_table.lookup(self.name)
+            except KeyError:
+                # TODO once #1258 is done the symbols should already exist
+                # and therefore we should raise an exception if not.
+                scalar_sym = symbol_table.new_symbol(
+                    self.name, symbol_type=DataSymbol,
+                    datatype=self.infer_datatype())
+            return scalar_sym
+
+        elif self.is_field:
+            # Although the argument to a Kernel is a field, the data itself
+            # is accessed through a field_proxy.
+            try:
+                sym = symbol_table.lookup(self.proxy_name)
+            except KeyError:
+                # TODO once #1258 is done the symbols should already exist
+                # and therefore we should raise an exception if not.
+                sym = symbol_table.new_symbol(
+                    self.proxy_name, symbol_type=DataSymbol,
+                    datatype=self.infer_datatype(proxy=True))
+            return sym
```

Comment:
Same here can you wrap it in a Reference even if then it is not used?
Also depending if fields are always passed as field%data to the kernels would it make sense to Create an StructureReference already here? This could help avoid repetition of logic about that a field is for LFRic but maybe there is some pragmatic reason not to do it? (e.g. I see in bultins you need structure of arrays accesses, can the indexing be easily added to an already contructed StructureReference)

Reply:
Yes, done. I couldn't easily change `psyir_expression` to give me back a StructureReference but I realised I could do the creation of that in (what was) `get_field_argument_symbols()` so I've done that.